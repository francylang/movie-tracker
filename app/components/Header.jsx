import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


const Header = ({ user, handleSignOut }) => {
  let userBox;

  const userContents = (
    <div className='user-box'>
      <p className='user'>Welcome, {user.name}!</p>
      <div className='two-user-btns'>
        <Link className='favorites-link' to='/favorites'>View Favorites</Link>
        <Link
          className='signout'
          to='/login'
          onClick={() => handleSignOut()}>
          Sign Out
        </Link>
      </div>
    </div>);

  const noUserContents =
  <div className='no-user-box'>
    <Link className='login-link' to='/login'>Log In</Link>
    <Link className='login-link' to='/createuser'>Create Account</Link>
  </div>;

  (!user.name) ? userBox = noUserContents : userBox = userContents;

  return (
    <div className='header'>
      <Link to='/'>
        <h1>
          <span className='header-one'>Movie</span>
          <span className='header-two'>Tracker</span>
        </h1>
      </Link>
      {userBox}
    </div>
  );
};

Header.propTypes = {
  user: PropTypes.object,
  handleSignOut: PropTypes.func
};

export default Header;
