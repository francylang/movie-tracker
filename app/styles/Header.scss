.header {
  position: fixed;
  top: 0;
  width: 100%;
  height: 120px;
  // background-color: #D23321;
  z-index: 100;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #0D0F0F;

  background-image: url("https://www.transparenttextures.com/patterns/carbon-fibre-v2.png");
}

.header h1{
  font-size: 60px;
  display: flex;
  margin-left: 75px;
}

.header-one {
  font-size: 75px;
  color: #F4F4F4;
}

.header-two {
  font-size: 75px;
  color: #D23321;
}

.no-user-box,
.user-box {
  display: flex;
  justify-content: center;
  margin-right: 75px;
}

.user {
  font-size: 20px;
  display: flex;
  align-items: center;
  padding-right: 200px;
}

.signout,
.favorites-link,
.login-link {
  display: flex;
  align-items: center;
  justify-content: center;
  color: #F4F4F4;
  text-decoration: none;
  padding: 5px 20px;
  margin: 0 10px;
  border: 1px solid #F4F4F4;
  border-radius: 10px;
}

.signout:hover,
.favorites-link:hover,
.login-link:hover {
  background-color: #D23321;
  border: none;
}
