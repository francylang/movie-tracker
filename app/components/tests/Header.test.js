import React from "react";
import Header from '../Header';
import { shallow } from "enzyme";

describe('Header component', () => {
  let wrapper = shallow(<Header user={{ name: 'Lola' }}/>);

  it('should render user-box is user signed in', () => {
    const userBox = wrapper.find('.user-box');

    expect(userBox.length).toEqual(1);
  });

  it('should render user name when user is signed in', () => {
    const name = wrapper.find('p');

    expect(name.text()).toEqual('Welcome, Lola!');
  });

  it('should render a link to view favorites', () => {
    const link = wrapper.find('Link');

    expect(link.length).toEqual(2);
  });

  it('should render a button to sign out', () => {
    const button = wrapper.find('.signout');

    expect(button.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  let altWrapper = shallow(<Header user={{}}/>);

  it('should render no-user-box if no user is signed in', () => {
    const noUserBox = altWrapper.find('.no-user-box');

    expect(noUserBox.length).toEqual(1);
  });

  it('should render 2 links', () => {
    const links = altWrapper.find('Link');

    expect(links.length).toEqual(2);
  });

  it('should match snapshot', () => {
    expect(altWrapper).toMatchSnapshot();
  });
});
