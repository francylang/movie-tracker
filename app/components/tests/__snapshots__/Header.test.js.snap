// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    user={
      Object {
        "name": "Lola",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
        >
          <h1>
            <span
              className="header-one"
            >
              Movie
            </span>
            <span
              className="header-two"
            >
              Tracker
            </span>
          </h1>
        </Link>,
        <div
          className="user-box"
        >
          <p
            className="user"
          >
            Welcome,
            Lola
            !
          </p>
          <Link
            className="login-link favorites-link"
            replace={false}
            to="/favorites"
          >
            View Favorites
          </Link>
          <Link
            className="login-link signout"
            onClick={[Function]}
            replace={false}
            to="/login"
          >
            Sign Out
          </Link>
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            <span
              className="header-one"
            >
              Movie
            </span>
            <span
              className="header-two"
            >
              Tracker
            </span>
          </h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="header-one"
              >
                Movie
              </span>,
              <span
                className="header-two"
              >
                Tracker
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Movie",
                "className": "header-one",
              },
              "ref": null,
              "rendered": "Movie",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Tracker",
                "className": "header-two",
              },
              "ref": null,
              "rendered": "Tracker",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="user"
            >
              Welcome,
              Lola
              !
            </p>,
            <Link
              className="login-link favorites-link"
              replace={false}
              to="/favorites"
            >
              View Favorites
            </Link>,
            <Link
              className="login-link signout"
              onClick={[Function]}
              replace={false}
              to="/login"
            >
              Sign Out
            </Link>,
          ],
          "className": "user-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome, ",
                "Lola",
                "!",
              ],
              "className": "user",
            },
            "ref": null,
            "rendered": Array [
              "Welcome, ",
              "Lola",
              "!",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "View Favorites",
              "className": "login-link favorites-link",
              "replace": false,
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "View Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Sign Out",
              "className": "login-link signout",
              "onClick": [Function],
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Sign Out",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            <h1>
              <span
                className="header-one"
              >
                Movie
              </span>
              <span
                className="header-two"
              >
                Tracker
              </span>
            </h1>
          </Link>,
          <div
            className="user-box"
          >
            <p
              className="user"
            >
              Welcome,
              Lola
              !
            </p>
            <Link
              className="login-link favorites-link"
              replace={false}
              to="/favorites"
            >
              View Favorites
            </Link>
            <Link
              className="login-link signout"
              onClick={[Function]}
              replace={false}
              to="/login"
            >
              Sign Out
            </Link>
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              <span
                className="header-one"
              >
                Movie
              </span>
              <span
                className="header-two"
              >
                Tracker
              </span>
            </h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="header-one"
                >
                  Movie
                </span>,
                <span
                  className="header-two"
                >
                  Tracker
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Movie",
                  "className": "header-one",
                },
                "ref": null,
                "rendered": "Movie",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Tracker",
                  "className": "header-two",
                },
                "ref": null,
                "rendered": "Tracker",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="user"
              >
                Welcome,
                Lola
                !
              </p>,
              <Link
                className="login-link favorites-link"
                replace={false}
                to="/favorites"
              >
                View Favorites
              </Link>,
              <Link
                className="login-link signout"
                onClick={[Function]}
                replace={false}
                to="/login"
              >
                Sign Out
              </Link>,
            ],
            "className": "user-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Welcome, ",
                  "Lola",
                  "!",
                ],
                "className": "user",
              },
              "ref": null,
              "rendered": Array [
                "Welcome, ",
                "Lola",
                "!",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "View Favorites",
                "className": "login-link favorites-link",
                "replace": false,
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "View Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Sign Out",
                "className": "login-link signout",
                "onClick": [Function],
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header component should match snapshot 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/"
        >
          <h1>
            <span
              className="header-one"
            >
              Movie
            </span>
            <span
              className="header-two"
            >
              Tracker
            </span>
          </h1>
        </Link>,
        <div
          className="no-user-box"
        >
          <Link
            className="login-link"
            replace={false}
            to="/login"
          >
            Log In
          </Link>
          <Link
            className="login-link"
            replace={false}
            to="/createuser"
          >
            Create Account
          </Link>
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            <span
              className="header-one"
            >
              Movie
            </span>
            <span
              className="header-two"
            >
              Tracker
            </span>
          </h1>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="header-one"
              >
                Movie
              </span>,
              <span
                className="header-two"
              >
                Tracker
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Movie",
                "className": "header-one",
              },
              "ref": null,
              "rendered": "Movie",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Tracker",
                "className": "header-two",
              },
              "ref": null,
              "rendered": "Tracker",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="login-link"
              replace={false}
              to="/login"
            >
              Log In
            </Link>,
            <Link
              className="login-link"
              replace={false}
              to="/createuser"
            >
              Create Account
            </Link>,
          ],
          "className": "no-user-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Log In",
              "className": "login-link",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Log In",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Create Account",
              "className": "login-link",
              "replace": false,
              "to": "/createuser",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/"
          >
            <h1>
              <span
                className="header-one"
              >
                Movie
              </span>
              <span
                className="header-two"
              >
                Tracker
              </span>
            </h1>
          </Link>,
          <div
            className="no-user-box"
          >
            <Link
              className="login-link"
              replace={false}
              to="/login"
            >
              Log In
            </Link>
            <Link
              className="login-link"
              replace={false}
              to="/createuser"
            >
              Create Account
            </Link>
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              <span
                className="header-one"
              >
                Movie
              </span>
              <span
                className="header-two"
              >
                Tracker
              </span>
            </h1>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="header-one"
                >
                  Movie
                </span>,
                <span
                  className="header-two"
                >
                  Tracker
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Movie",
                  "className": "header-one",
                },
                "ref": null,
                "rendered": "Movie",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Tracker",
                  "className": "header-two",
                },
                "ref": null,
                "rendered": "Tracker",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="login-link"
                replace={false}
                to="/login"
              >
                Log In
              </Link>,
              <Link
                className="login-link"
                replace={false}
                to="/createuser"
              >
                Create Account
              </Link>,
            ],
            "className": "no-user-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Log In",
                "className": "login-link",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Log In",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Create Account",
                "className": "login-link",
                "replace": false,
                "to": "/createuser",
              },
              "ref": null,
              "rendered": "Create Account",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
